controllers/cluster_controller.go

```diff
@@ -1097,6 +1097,39 @@ type localityInfo struct {
 
 	// The locality map.
 	LocalityData map[string]string
+
+	Class fdbtypes.ProcessClass
+}
+
+// These indexes are used for sorting and since we sort ascending
+func getClassIndex(cls fdbtypes.ProcessClass) int {
+	switch cls {
+	case fdbtypes.ProcessClassStorage:
+		return 0
+	case fdbtypes.ProcessClassLog:
+		return 1
+	case fdbtypes.ProcessClassTransaction:
+		return 2
+	}
+
+	return math.MaxInt32
+}
+
+// This will sort the processes according to the following rules:
+// First all storage processes next log processes and then tlog processes.
+// Inside each process class the processes will be sorted by the ID (lower IDs come first).
+// We have to do this to ensure we get a deterministic result for selecting the candidates
+// otherwise we get a (nearly) random result since processes are stored in a map which is by definition
+// not sorted and doesn't return values in a stable way.
+func sortLocalities(processes []localityInfo) {
+	// Sort the processes for ID to ensure we have a stable input
+	sort.SliceStable(processes, func(i, j int) bool {
+		if processes[i].Class == processes[j].Class {
+			return processes[i].ID < processes[j].ID
```

Comment:
This will sort alphanumeric, we could also sort only based on the instance ID number but I don't think that the additional effort adds anything.

Reply:
I agree.